pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/OpenTelemetryConsumerStats.java

```diff
@@ -0,0 +1,170 @@
+                .setDescription("The total number of messages dispatched to this consumer.")
+                .buildObserver();
+
+        bytesOutCounter = meter
+                .counterBuilder(BYTES_OUT_COUNTER)
+                .setUnit("By")
+                .setDescription("The total number of messages bytes dispatched to this consumer.")
+                .buildObserver();
+
+        messageAckCounter = meter
+                .counterBuilder(MESSAGE_ACK_COUNTER)
+                .setUnit("{ack}")
+                .setDescription("The total number of message acknowledgments received from this consumer.")
+                .buildObserver();
+
+        messageRedeliverCounter = meter
+                .counterBuilder(MESSAGE_REDELIVER_COUNTER)
+                .setUnit("{message}")
+                .setDescription("The total number of messages that have been redelivered to this consumer.")
+                .buildObserver();
+
+        messageUnacknowledgedCounter = meter
+                .upDownCounterBuilder(MESSAGE_UNACKNOWLEDGED_COUNTER)
+                .setUnit("{message}")
+                .setDescription("The total number of messages unacknowledged by this consumer.")
+                .buildObserver();
+
+        messagePermitsCounter = meter
+                .upDownCounterBuilder(MESSAGE_PERMITS_COUNTER)
+                .setUnit("{permit}")
+                .setDescription("The number of permits currently available for this consumer.")
+                .buildObserver();
+
+        batchCallback = meter.batchCallback(() -> pulsar.getBrokerService()
+                        .getTopics()
+                        .values()
+                        .stream()
+                        .map(topicFuture -> topicFuture.getNow(Optional.empty()))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .map(Topic::getSubscriptions)
+                        .flatMap(s -> s.values().stream())
+                        .map(Subscription::getConsumers)
+                        .flatMap(Collection::stream)
+                        .forEach(this::recordMetricsForConsumer),
+                messageOutCounter,
+                bytesOutCounter,
+                messageAckCounter,
+                messageRedeliverCounter,
+                messageUnacknowledgedCounter,
+                messagePermitsCounter);
+    }
+
+    @Override
+    public void close() {
+        batchCallback.close();
+    }
+
+    private void recordMetricsForConsumer(Consumer consumer) {
+        var subscription = consumer.getSubscription();
+        var topicName = TopicName.get(subscription.getTopic().getName());
+
+        var builder = Attributes.builder()
+                .put(OpenTelemetryAttributes.PULSAR_CONSUMER_NAME, consumer.consumerName())
+                .put(OpenTelemetryAttributes.PULSAR_CONSUMER_ID, consumer.consumerId())
+                .put(OpenTelemetryAttributes.PULSAR_CONSUMER_CONNECTED_SINCE,
```

Comment:
@merlimat @dragosvictor This and other attributes here are super high cardinality - why are we recording with them? This will "kill" any TSDB and be very costly. Even when compared with `pulsar_consumer_msg_rate_out` - it didn't have all those attributes.